packages/slice-machine/components/Simulator/components/Header/index.tsx

```diff
@@ -1,20 +1,36 @@
 import router from "next/router";
 import { Box, Text, Flex } from "theme-ui";
 import * as Models from "@slicemachine/core/build/models";
-
+import useSliceMachineActions from "src/modules/useSliceMachineActions";
 import VarationsPopover from "@lib/builders/SliceBuilder/Header/VariationsPopover";
 import * as Links from "@lib/builders/SliceBuilder/links";
 
 import ScreenSizes, { Size } from "../ScreenSizes";
 import { ComponentUI } from "@lib/models/common/ComponentUI";
+import { useState } from "react";
+import SaveButton from "@lib/builders/SliceBuilder/Header/SaveButton";
 
 type PropTypes = {
   Model: ComponentUI;
-  variation: Models.VariationSM | undefined;
+  variation: Models.VariationSM;
```

Comment:
Was there a reason for this to be optional before? Didn't seem to have broken anything afaik

Reply:
I believe it might come from the fact that the store "SelectedSlice" might be undefined at first before being updated.